src/shared/tests.h

```diff
@@ -187,6 +187,7 @@ static inline int run_test_table(void) {
                         _r = run_test_table();            \
                 _q = _outro ? _outro() : EXIT_SUCCESS;    \
                 static_destruct();                        \
+                hashmap_trim_pools();                     \
```

Comment:
i think in the tests it makes a lot of sense to do this always, hence this addition does make a ton of sense to me.

Reply:
Maybe. But tentatively dropped, now this change is unrelated to this PR.